js/search/search.js

```diff
@@ -178,57 +158,72 @@ class Search {
      * An array of messages to be indexed
      * in real time
      * @param message
+     * @param callback
      */
-    realTimeIndexing(message) {
+    realTimeIndexing(message, callback) {
+        if (typeof callback !== "function") {
+            return false;
+        }
 
         try {
             let msg = JSON.parse(message);
             if (!(msg instanceof Array)) {
-                log.send(logLevels.ERROR, 'RealTime Indexing: Messages must be an array real-time indexing');
-                return (new Error('RealTime Indexing: Messages must be an array'));
+                return callback(false, 'RealTime Indexing: Messages must be an array');
```

Comment:
Can we abstract out all this logic to a separate method? Let's avoid redundancy.